builds/e2e/longhaul.yaml

```diff
@@ -349,4 +349,67 @@ jobs:
           snitch.storage.account: '$(snitch.storage.account)'
           snitch.storage.masterKey: '$(StorageAccountMasterKeyStress)'
           loadGen.message.frequency: '$(loadGen.message.frequency.amd64)'
+
+################################################################################
+  - job: windows_servercore_amd64_moby
+################################################################################
+    displayName: Windows Server Core AMD64 Moby
+    condition: eq(variables['run.windows.amd64.moby'], 'true')
+    pool:
+      name: $(pool.name)
+      demands:
+        - Agent.OSName -equals ServerCore
```

Comment:
nit: suggest "WindowsServerCore" or "WinServerCore"

Reply:
nit: suggest to use "agent-osname" in order to distinguish between auto-generated and user-defined.

Reply:
please also add "agent-osname" demands in existing Windows 10 pro x64 explicitly for both stress and long haul.